flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamGraphGeneratorTest.java

```diff
@@ -749,6 +750,46 @@ public void testSettingSavepointRestoreSettingsSetterOverrides() {
                 equalTo(SavepointRestoreSettings.forPath("/tmp/savepoint1")));
     }
 
+    @Test
+    public void testConfigureSlotSharingGroupResource() {
+        final SlotSharingGroup ssg1 =
+                SlotSharingGroup.newBuilder("ssg1").setCpuCores(1).setTaskHeapMemoryMB(100).build();
+        final SlotSharingGroup ssg2 =
+                SlotSharingGroup.newBuilder("ssg2").setCpuCores(2).setTaskHeapMemoryMB(200).build();
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+        final DataStream<Integer> source = env.fromElements(1).slotSharingGroup("ssg1");
+        source.map(value -> value)
+                .slotSharingGroup(ssg2)
+                .addSink(new DiscardingSink<>())
+                .slotSharingGroup(ssg1);
+
+        final StreamGraph streamGraph = env.getStreamGraph();
+        assertThat(
+                streamGraph.getSlotSharingGroupResource("ssg1").get(),
+                is(ResourceProfile.fromResources(1, 100)));
+        assertThat(
```

Comment:
Better to also include
1. a transformation without `SlotSharingGroup` set
2. a transformation with "default" `SlotSharingGroup` set